ktor-client/ktor-client-curl/desktop/src/io/ktor/client/engine/curl/internal/CurlMultiApiHandler.kt

```diff
@@ -130,27 +130,22 @@ internal class CurlMultiApiHandler : Closeable {
     }
 
     @OptIn(ExperimentalForeignApi::class)
-    internal fun perform() {
+    internal fun perform(transfersRunning: IntVarOf<Int>) {
         if (activeHandles.isEmpty()) return
 
-        memScoped {
-            val transfersRunning = alloc<IntVar>()
-            do {
-                synchronized(easyHandlesToUnpauseLock) {
-                    var handle = easyHandlesToUnpause.removeFirstOrNull()
-                    while (handle != null) {
-                        curl_easy_pause(handle, CURLPAUSE_CONT)
-                        handle = easyHandlesToUnpause.removeFirstOrNull()
-                    }
-                }
-                curl_multi_perform(multiHandle, transfersRunning.ptr).verify()
-                if (transfersRunning.value != 0) {
-                    curl_multi_poll(multiHandle, null, 0.toUInt(), 10000, null).verify()
-                }
-                if (transfersRunning.value < activeHandles.size) {
-                    handleCompleted()
-                }
-            } while (transfersRunning.value != 0)
+        synchronized(easyHandlesToUnpauseLock) {
+            var handle = easyHandlesToUnpause.removeFirstOrNull()
+            while (handle != null) {
+                curl_easy_pause(handle, CURLPAUSE_CONT)
+                handle = easyHandlesToUnpause.removeFirstOrNull()
+            }
+        }
+        curl_multi_perform(multiHandle, transfersRunning.ptr).verify()
+        if (transfersRunning.value != 0) {
+            curl_multi_poll(multiHandle, null, 0.toUInt(), 100, null).verify()
```

Comment:
Is this timeout value okay? It directly affects how fast HTTP request will be sent when we have established websocket connection so I've reduced it from 10 to 0.1 sec.